### PR TITLE
return the last error with any error we get

### PR DESCRIPTION
**Pull request by asalkeld**
/cc @stbenjam

### CODE REVIEW

File: ironic/workflow.go [hunk 1]
```diff
@@ -100,25 +102,23 @@ func (workflow *provisionStateWorkflow) toManageable() (done bool, err error) {
...
-
-	return false, nil
 }
 
 // Clean a node
-func (workflow *provisionStateWorkflow) toClean() (done bool, err error) {
+func (workflow *provisionStateWorkflow) toClean() (bool, error) {
 	// Node must be manageable first
-	workflow.reloadNode()
+	_ = workflow.reloadNode()
```
Comment (stbenjam) [inline ironic/workflow.go:110]: reloadNode returns an error, what do you think about checking them? Same for changeProvisionState.  Sorry this code is some of the first go I ever wrote.   It's probably pretty sketchy.

Comment (asalkeld) [reply ironic/workflow.go:110]: sure thing

File: ironic/data_source_ironic_introspection.go [hunk 2]
```diff
@@ -79,11 +80,26 @@ func dataSourceIronicIntrospectionRead(d *schema.ResourceData, meta interface{})
...
-	d.Set("finished", status.Finished)
-	d.Set("finished_at", status.FinishedAt)
-	d.Set("started_at", status.StartedAt)
-	d.Set("error", status.Error)
-	d.Set("state", status.State)
+	err = d.Set("finished", status.Finished)
+	if err != nil {
+		return err
+	}
+	err = d.Set("finished_at", status.FinishedAt.Format("2006-01-02T15:04:05"))
```
Comment (asalkeld) [inline ironic/data_source_ironic_introspection.go:87]: this never worked as there was a type error (setting time.Time to a string field)

File: ironic/resource_ironic_node_v1.go [hunk 3]
```diff
@@ -371,11 +435,12 @@ func resourceNodeV1Update(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	// Clean node
```
Comment (asalkeld) [inline ironic/resource_ironic_node_v1.go:437]: I am not sure what to do about this.. In bmo we look at hardware info to check for raid cleaning.. not sure if this is needed, maybe just remove this?

Comment (stbenjam) [reply ironic/resource_ironic_node_v1.go:437]: If it's never worked, we could just remove it unless we can do the right thing based on the hardware info? It probably makes sense that we should actually clean these nodes

Comment (asalkeld) [reply ironic/resource_ironic_node_v1.go:437]: so bmo needs the bmc drivers to set the raid config, I am not sure we can do that?
https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/ironic.go#L1249-L1292
Also it probably should be done in a separate PR, I'll make it obvious and we can think if we need to raise an issue for it?

**Comment by stbenjam** [comment]
This looks good to me, thanks

**Pull request merged**